core/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -247,23 +245,71 @@ public XContentBuilder toXContentEmbedded(XContentBuilder builder, Params params
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
         if (!isExists()) {
-            builder.field(Fields._INDEX, index);
-            builder.field(Fields._TYPE, type);
-            builder.field(Fields._ID, id);
-            builder.field(Fields.FOUND, false);
+            builder.field(_INDEX, index);
+            builder.field(_TYPE, type);
+            builder.field(_ID, id);
+            builder.field(FOUND, false);
         } else {
-            builder.field(Fields._INDEX, index);
-            builder.field(Fields._TYPE, type);
-            builder.field(Fields._ID, id);
+            builder.field(_INDEX, index);
+            builder.field(_TYPE, type);
+            builder.field(_ID, id);
             if (version != -1) {
-                builder.field(Fields._VERSION, version);
+                builder.field(_VERSION, version);
             }
             toXContentEmbedded(builder, params);
```

Comment:
this object is reused in quite a few places where we call this `toXContentEmbedded` method to print out only part of it within a different ancestor. I am not sure a new interface would help this specific case, cause this inner method is not just about starting the ancestor externally, but also printing out only part of the object. it may help other cases though.